gym/spaces/box.py

```diff
@@ -44,7 +45,7 @@ def __init__(
         high: Union[SupportsFloat, np.ndarray],
         shape: Optional[Sequence[int]] = None,
         dtype: Type = np.float32,
-        seed: Optional[int] = None,
+        seed: Optional[int | seeding.RandomNumberGenerator] = None,
```

Comment:
Isn't this a bit too strict of a requirement? Couldn't one pass a `np.random.Generator` here?

Reply:
As seeding.RandomNumberGenerator is a subclass of np.random.Generator and in the future, seeding.RandomNumberGenerator is planned to be removed (there isn't a planned date but I think Gym 1.0 would be the plan).
I can reduce the strictness but this is the current gym randomness method

Reply:
Why not allow a `np.random.RandomState`?

Reply:
I guess this is probably out of scope for this PR, so no need to change this, but I'm really skeptical about the usefulness of having this `seeding.RandomNumberGenerator` thingy. If someone can point me to the rationale behind using it, I'd be curious to take a look.

Reply:
>If someone can point me to the rationale behind using it, I'd be curious to take a look.

It's a [compatibility layer](https://github.com/openai/gym/blob/master/gym/utils/seeding.py#L23) that supports both the new `np.random.Generator` API as well as the older `np.random.RandomState` which is now considered legacy